src/sagas/bonding-curve.js

```diff
@@ -0,0 +1,363 @@
+import ContractImplementation from 'kleros-api-2/lib/contracts/ContractImplementation'
+
+import { takeLatest, call, select, all, put } from 'redux-saga/effects'
+
+import uniswapArtifact from '../assets/contracts/Uniswap'
+import ERC20Artifact from '../assets/contracts/ERC20'
+import { eth, BONDING_CURVE_ADDRESS } from '../bootstrap/dapp-api'
+import * as bondingCurveActions from '../actions/bonding-curve'
+import * as arbitratorActions from '../actions/arbitrator'
+import * as walletActions from '../actions/wallet'
+import * as walletSelectors from '../reducers/wallet'
+import { lessduxSaga } from '../utils/saga'
+
+/**
+ * The factory method for accessing the bonding curve contract. This method is
+ * necessary because the contract address is not available until the <Initializer>
+ * mounts, so we can't bind a module level variable to a BondingCurve instance.
+ * @returns {BondingCurve} The BondingCurve instance.
+ */
+const getBondingCurve = (function() {
```

Comment:
I think this is unnecessary. You don't have any root sagas. So no saga will be called before initializer mounts. You could initialize and export `BondingCurve` in bootstrap similar to how `Kleros` is initialized and exported. Is there a reason you would want to re-initialize the object every call?

Reply:
Will change.